hexrd/material.py

```diff
@@ -576,7 +576,10 @@ def _readCif(self, fcif=DFLT_NAME+'.cif'):
 
             for ii, x in enumerate(chkstr):
                 if x:
-                    U[ii] = 1.0/numpy.pi/2./numpy.sqrt(2.)
+                    try:
+                        U[ii] = float(U[ii])
+                    except:
+                        U[ii] = self.DFLT_U[0]
```

Comment:
It's usually best to avoid bare `except` since it will literally catch anything, including errors that we would want to know about (which can lead to bugs that are hard to track down).

What are the kinds of errors that can occur here? `IndexError` from the `ii` index being out of range? We could potentially catch that instead. Or there could also be a `ValueError` converting to float?

Reply:
`U[ii]`, when read from the file returns a string. Now, this string could be a number like `0.0` or a `?` to denote that the number is unavailable. The try was to make sure `float(U[ii])` was successful, and if it was not then we can just assign the default value. Can you think of a better way to do this?

Reply:
We could do `if U[ii] == '?'` to check if it is a question mark, if the only allowable non-float value is a question mark (that way, if some typo is entered instead, it would produce an error).

Or, the exception could catch `ValueError` to catch any errors when converting to float.